main.tf

```diff
@@ -2758,7 +2758,7 @@ module "karpenter" {
       },
       {
         name  = "settings.aws.interruptionQueueName"
-        value = module.karpenter_sqs.queue_name
+        value = local.karpenter_enable_spot_termination ? module.karpenter_sqs.queue_name : ""
       },
```

Comment:
Do you think is better to skip the whole block than passing an empty string?
```hcl
},
       local.karpenter_enable_spot_termination ? {
         name  = "settings.aws.interruptionQueueName"
         value = module.karpenter_sqs.queue_name
        } : {}
,
{
```

Reply:
That doesn't work. The helm provider is expecting maps in set list to have name and value keys.

```
Error: Missing map element
│ 
│   on .terraform/modules/eks_blueprints_addons.karpenter/main.tf line 60, in resource "helm_release" "this":
│   60:       name  = set.value.name
│     ├────────────────
│     │ set.value is empty map of string
│ 
│ This map does not have an element with the key "name".
╵
╷
│ Error: Missing map element
│ 
│   on .terraform/modules/eks_blueprints_addons.karpenter/main.tf line 61, in resource "helm_release" "this":
│   61:       value = set.value.value
│     ├────────────────
│     │ set.value is empty map of string
│ 
│ This map does not have an element with the key "value".
```

Reply:
if you wanted to go this route, I think you would need to:
1. Use `compact()` such as on line 2745 would become `set = compact(concat(`
2. Move the conditional inside the map like:
```hcl
      { for k,v in {
           name  = "settings.aws.interruptionQueueName"
           value = module.karpenter_sqs.queue_name
        } k => v if local.karpenter_enable_spot_termination
      }
```

But since Helm treats an empty map value as a no-op, I think thats the easiest route for now